hack/make-rules/test-cmd-util.sh

```diff
@@ -3041,6 +3044,58 @@ run_daemonset_history_tests() {
   set +o errexit
 }
 
+run_statefulset_history_tests() {
+  set -o nounset
```

Comment:
Leave this set, and if you want to not fail on certain calls them put them in an "if"

Reply:
i follow the template of `run_daemonset_history_tests`: https://github.com/kubernetes/kubernetes/blob/master/hack/make-rules/test-cmd-util.sh#L2998

If we are to remove this line for statefulset, we should also do so for daemonset.

Reply:
Ok